common/djangoapps/student/scope_resolver.py

```diff
@@ -57,6 +57,65 @@ def resolve(self, scope_name, scope_context, instance_context):
         )
 
 
+class NamespaceEnrollmentsScopeResolver(NotificationUserScopeResolver):
+    """
+    Implementation of the NotificationUserScopeResolver abstract
+    interface defined in edx-notifications.
+
+    We will be passed in a namespace (aka course_id) in the context
+    and we must return a Django ORM resultset or None if
+    we cannot match.
+    """
+
+    def resolve(self, scope_name, scope_context, instance_context):
+        """
+        The entry point to resolve a scope_name with a given scope_context
+        """
+
+        if scope_name != 'namespace_scope':
+            # we can't resolve any other scopes
+            return None
+
+        if 'namespace' not in scope_context:
+            # did not receive expected parameters
+            return None
+
+        course_id = scope_context['namespace']
+
+        if not isinstance(course_id , CourseKey):
+            try:
```

Comment:
I feel like this course key generation logic with the try/except and SlashSeparatedCourseKey thing is repeated all over the place -- is there a common module/method that can be called instead?